java/vector/src/main/java/org/apache/arrow/vector/compare/RangeEqualsVisitor.java

```diff
@@ -450,6 +454,77 @@ protected boolean compareBaseLargeVariableWidthVectors(Range range) {
     return true;
   }
 
+  protected boolean compareBaseVariableWidthViewVectors(Range range) {
+    BaseVariableWidthViewVector leftVector = (BaseVariableWidthViewVector) left;
+    BaseVariableWidthViewVector rightVector = (BaseVariableWidthViewVector) right;
+
+    final ArrowBuf leftViewBuffer = leftVector.getDataBuffer();
+    final ArrowBuf rightViewBuffer = rightVector.getDataBuffer();
+
+    final int elementSize = BaseVariableWidthViewVector.ELEMENT_SIZE;
+    final int lengthWidth = BaseVariableWidthViewVector.LENGTH_WIDTH;
+    final int prefixWidth = BaseVariableWidthViewVector.PREFIX_WIDTH;
+    final int bufIndexWidth = BaseVariableWidthViewVector.BUF_INDEX_WIDTH;
+
+    for (int i = 0; i < range.getLength(); i++) {
+      int leftIndex = range.getLeftStart() + i;
+      int rightIndex = range.getRightStart() + i;
+
+      boolean isNull = leftVector.isNull(leftIndex);
+      if (isNull != rightVector.isNull(rightIndex)) {
+        return false;
+      }
+
+      int startIndexLeft = leftIndex * elementSize;
+      int endIndexLeft = (leftIndex + 1) * elementSize;
+
+      int startIndexRight = rightIndex * elementSize;
+      int endIndexRight = (rightIndex + 1) * elementSize;
+
+      // check equality of the view
+      int retView = ByteFunctionHelpers.equal(leftViewBuffer, startIndexLeft, endIndexLeft,
+          rightViewBuffer, startIndexRight, endIndexRight);
```

Comment:
Why are we comparing all 16 bytes? Or rather, why are we comparing the raw bytes here at all?

Reply:
I think we need to check the length, and then compare the remaining 12 bytes or compare the data buffers.

Reply:
Yes. Good point.